File: client/csharp/example/quick/Finger.cs
using System;
using System.IO;
using Finger;

namespace example
{
    class FingerTest
    {
        private const string FINGERPRINT_IMAGE_FILE = "./finger.bmp";
        private const uint QUALITY_THRESHOLD = 50;

        private FingerSvc fingerSvc;

        public FingerTest(FingerSvc svc) {
            fingerSvc = svc;
        }

        public void Test(uint deviceID) {
            Console.WriteLine(">>> Scan a finger...");

            var templateData = fingerSvc.Scan(deviceID, Finger.TemplateFormat.Suprema, QUALITY_THRESHOLD);

            Console.WriteLine("Finger template: {0}" + Environment.NewLine, BitConverter.ToString(templateData.ToByteArray()));

            var bmpImage = fingerSvc.GetImage(deviceID);
            File.WriteAllBytes(FINGERPRINT_IMAGE_FILE, bmpImage.ToByteArray());

            var fingerConfig = fingerSvc.GetConfig(deviceID);

            Console.WriteLine("Finger config: {0}" + Environment.NewLine, fingerConfig);
        }        
    }
}

File: client/go/src/example/quick/finger.go
package main

import (
	"biostar/service/finger"
	"io/ioutil"
	"fmt"
)

const (
	QUALITY_THRESHOLD uint32 = 50
	FINGERPRINT_IMAGE_NAME = "./finger.bmp"
)

func testFinger(deviceID uint32) error {
	fmt.Printf(">>> Scan a finger...\n")

	templData, score, err := fingerSvc.Scan(deviceID, finger.TemplateFormat_TEMPLATE_FORMAT_SUPREMA, QUALITY_THRESHOLD)

	if err != nil {
		return err
	}

	fmt.Printf("Template: %v %v\n\n", score, templData)

	bmpImage, err := fingerSvc.GetImage(deviceID)

	if err != nil {
		return err
	}

	ioutil.WriteFile(FINGERPRINT_IMAGE_NAME, bmpImage, 0644)

	fingerConfig, err := fingerSvc.GetConfig(deviceID)

	if err != nil {
		return err
	}

	fmt.Printf("Fingerprint config: %v\n\n", fingerConfig)

	return nil
}


